integration/containerd/confidential/agent_api.bats

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bats
+# Copyright (c) 2022 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/../../../lib/common.bash"
+
+setup() {
+	sandbox_name="kata-cc-busybox-sandbox"
+	pod_config="${FIXTURES_DIR}/pod-config.yaml"
+        pod_id=""
+
+	echo "Delete any existing ${sandbox_name} pod"
+	crictl_delete_cc_pod_if_exists "$sandbox_name"
+
+	# TODO: if I create the CC handler for containerd and not offload
+	#       image service then the container creation fail. Maybe this
+	#	test should be merged on main directly, and avoid using
+	#	CC handler. Otherwise there will be needed changes.
+
+	#echo "Prepare containerd for Confidential Container"
+	#SAVED_CONTAINERD_CONF_FILE="/etc/containerd/config.toml.$$"
+	#configure_cc_containerd "$SAVED_CONTAINERD_CONF_FILE"
+
+	enable_full_debug
+}
+
+@test "[cc][agent][api][cri][containerd] Test agent API endpoint can be restricted" {
+	local agent_config_filename="agent-configuration-no-exec.toml"
+
+	cp_to_guest_img "/tests/fixtures" \
+		"${FIXTURES_DIR}/${agent_config_filename}"
+
+	add_kernel_params \
+		"agent.config_file=/tests/fixtures/${agent_config_filename}"
+
+	# The endpoint "ExecProcessRequest" is not allowed.
+	crictl_create_cc_pod "$pod_config"
+
+	local container_config="${FIXTURES_DIR}/container-config.yaml"
+
+	crictl_create_cc_container "$sandbox_name" "$pod_config" \
+		"$container_config"
+
+	local pod_id=$(crictl pods --name "$sandbox_name" -q)
+        local container_id=$(crictl ps --pod ${pod_id} -q)
+	# The endpoint "ExecProcessRequest" is not allowed so this exec
+	# operation should fail
+	! crictl exec "$container_id" cat /proc/cmdline
+
+	# TODO: grep the logs to be sure if failed because of the blocked endpoint.
```

Comment:
@wainersm Are you planning to add this check for logs ?

Reply:
Hi @amshinde , yes, I will lift the rfc label and send a new version of this PR where I am going to introduce that check.